src/main/java/seedu/address/model/transaction/Quantity.java

```diff
@@ -7,29 +7,101 @@
  * Class to store the quantity of goods transacted.
  */
 public class Quantity {
-    public static final String MESSAGE_CONSTRAINTS =
-            "Quantity should only contain numbers, and it should be at least 1 digits long";
+
     public static final String VALIDATION_REGEX = "\\d{1,}";
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Quantity should only contain numbers and no decimal point.";
+
+    public static final String MESSAGE_CONSTRAINTS_ZERO =
+            "Quantity should not be 0.";
+    public static final String MESSAGE_CONSTRAINTS_POSITIVE =
+            "Quantity should be not be negative.";
+
+    public static final String MESSAGE_CONSTRAINTS_LARGE =
+            "Quantity should be not be more than 1 million.";
+
+    public static final String MESSAGE_CONSTRAINTS_EMPTY = "Quantity should not be left empty.";
+
     public final String quantity;
 
     /**
      * Constructs a {@code Quantity}.
-     *
      * @param quantity A valid quantity.
      */
     public Quantity(String quantity) {
         requireNonNull(quantity);
+
+        checkArgument(isValidQuantity_empty(quantity), MESSAGE_CONSTRAINTS_EMPTY);
         checkArgument(isValidQuantity(quantity), MESSAGE_CONSTRAINTS);
-        this.quantity = quantity;
+        checkArgument(isPositiveQuantity(quantity), MESSAGE_CONSTRAINTS_POSITIVE);
+        checkArgument(isValidQuantity_regex(quantity), MESSAGE_CONSTRAINTS);
+        checkArgument(isSmallQuantity(quantity), MESSAGE_CONSTRAINTS_LARGE);
+        checkArgument(isValidQuantity_nonZero(quantity), MESSAGE_CONSTRAINTS_ZERO);
+        this.quantity = Integer.toString(Integer.parseInt(quantity));
     }
 
     /**
      * Returns true if a given string is a valid quantity.
      */
+
     public static boolean isValidQuantity(String test) {
+        requireNonNull(test);
+        boolean isInteger = true;
```

Comment:
You don't need isInteger, just return false at line 54 and return true at line 57